### PR TITLE
[multiple] Remove custom analysis_options.yaml from web plugins.

### PR DESCRIPTION
**Pull request by ditman**
## Description

Substitute `undefined_prefixed_name: ignore` analyzer setting by a `dart:ui` shim with conditional exports for the following plugins:

* video_player_web
* google_maps_flutter_web
* url_launcher_web

## Related Issues

* https://github.com/flutter/flutter/issues/41563 (root cause)
* https://github.com/flutter/flutter/issues/55000 (cleanup ticket that will be repurposed for this new approach)
* https://github.com/flutter/flutter/issues/69309

## video_player_web tests

Video player web tests are broken (with nnbd errors), but so are in `master`. I'll migrate them to run with `flutter drive` instead of `flutter test --platform chrome`, which is being deprecated.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/shims/dart_ui.dart [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// This file shims dart:ui in web-only scenarios, getting rid of the need to
+/// suppress analyzer warnings.
+
+export 'dart_ui_fake.dart' if (dart.library.html) 'dart_ui_real.dart';
```
Comment (amirh) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/shims/dart_ui.dart:8]: Do we want a TODO to remove this with a link to https://github.com/flutter/flutter/issues/55000 ?

**Review by amirh — APPROVED** [review]
Thanks! LGTM

If we see this as a temporary workaround we should probably have TODOs to remove it on all files. If it's a permanent solution we should probably put the workaround in a common package.

**Comment by amirh** [comment]
Note that the broken analyzer is due to null safety being turned on by default, see discussion on: https://github.com/flutter/flutter/pull/69440#issuecomment-719877525

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/shims/dart_ui.dart (hunk 1 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/shims/dart_ui.dart:8]: Yes, I'll add a TODO to 55000, this change doesn't fix the underlying issue yet.

**Comment by ditman** [comment]
Rebased on top of the latest master.

**Pull request merged**